coordinator/src/main/scala/filodb.coordinator/queryplanner/LogicalPlanUtils.scala

```diff
@@ -157,4 +161,38 @@ object LogicalPlanUtils {
     } else {
       labels
     }
+
+  /**
+    * Split query if the time range is greater than the threshold. It clones the given LogicalPlan with the smaller
+    * time ranges, creates an execPlan using the provided planner and finally returns Stitch ExecPlan
+    * @param lPlan LogicalPlan to be split
+    * @param qContext QueryContext
+    */
+  def splitPlans(lPlan: LogicalPlan,
+                 qContext: QueryContext,
+                 timeSplitEnabled: Boolean,
+                 splitThresholdMs: Long,
+                 splitSizeMs: Long): Seq[LogicalPlan] = {
+    val lp = lPlan.asInstanceOf[PeriodicSeriesPlan]
+    if (timeSplitEnabled && lp.isSplittable && lp.endMs - lp.startMs > splitThresholdMs && lp.stepMs <= splitSizeMs) {
+      logger.info(s"Splitting query queryId=${qContext.queryId} start=${lp.startMs}" +
+        s" end=${lp.endMs} step=${lp.stepMs} splitThresholdMs=$splitThresholdMs splitSizeMs=$splitSizeMs")
+      val windowSplits: Buffer[(Long, Long)] = Buffer.empty
```

Comment:
Just use ArrayBuffer which is already a Seq. debox benefit  is only when native types are used.

Reply:
ack, fixed it